### ISSUE TITLE
Review -- BiGAN: Adversarial Feature Learning (GAN).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#53]
Sik-Ho Tsang. [Review — BiGAN: Adversarial Feature Learning (GAN)](https://medium.com/nerd-for-tech/review-bigan-adversarial-feature-learning-gan-535eb76be2ca).

**Comment by NorbertZheng** [comment]
## Overview
Bidirectional Generative Adversarial Networks (BiGANs): Learning the Inverse Mapping, from Image Space to Latent Space.

In this story, Adversarial Feature Learning, (BiGAN), by the University of California, and the University of Texas, is briefly reviewed. In this paper:
 - Bidirectional Generative Adversarial Network (BiGAN) is designed as a means of learning inverse mapping, i.e. projecting data $x$ back into the latent space $z$.
 - This resulting learned feature representation is useful for **auxiliary supervised discrimination tasks**.

This is a paper in 2017 ICLR with over 1100 citations.

The idea is the same as [ALI](https://sh-tsang.medium.com/review-ali-adversarially-learned-inference-gan-6e4677667914) while they are proposed independently and published in the same conference (2017 ICLR). Some papers would cite both BiGAN and [ALI](https://sh-tsang.medium.com/review-ali-adversarially-learned-inference-gan-6e4677667914) together when talking about this idea.

**Comment by NorbertZheng** [comment]
## BiGAN: Overall Structure
![image](https://user-images.githubusercontent.com/37235023/216498052-88922715-ed0b-42e9-851c-d6540dce0a92.png)
BiGAN: Overall Structure.
 - In addition to the generator $G$ from the standard [GAN](https://medium.com/@sh.tsang/review-gan-generative-adversarial-nets-gan-e12793e1fb75), BiGAN includes an encoder $E$ which maps data $x$ to latent representations $z$.
 - The BiGAN discriminator $D$ discriminates not only in data space ( $x$ versus $G(z)$ ) but jointly in data and latent space (tuples $(x, E(x))$ versus $(G(z), z))$, where the latent component is either an encoder output $E(x)$ or a generator input $z$.
 - The data space is flattened as a vector and concatenated with the latent space vector, then input into the discriminator $D$.
 - In this context, a latent representation $z$ may be thought of as a “label” for $x$, but one which came for “free,” without the need for supervision.
 - The BiGAN training objective is defined as a minimax objective:

$$
\min_{G,E}\max_{D}V(D,E,G)
$$

$$
V(D,E,G):=\mathbb{E}_{x \sim p(x)}[\underbrace{\mathbb{E}_{z \sim p_{E}(\cdot|x)}[logD(x,z)]}_{logD(x,E(x))}]+\mathbb{E}_{z \sim p(z)}[\underbrace{\mathbb{E}_{x \sim p_{G}(\cdot|z)}[log(1-D(x,z))]}_{log(1-D(G(z),z))}].
$$

 - The same alternating gradient based optimization as [GAN](https://medium.com/@sh.tsang/review-gan-generative-adversarial-nets-gan-e12793e1fb75) is used.
 - In one iteration, the discriminator parameters $\theta D$ are updated by taking one or more steps **in the positive gradient direction**.
 - Then, the encoder parameters $\theta E$ and generator parameters $\theta G$ are together updated by taking a step in **the negative gradient direction**.

A model trained to predict features $z$ given data $x$ should learn useful semantic representations. BiGAN objective forces the encoder $E$ to do exactly this.

In order to fool the discriminator at a particular $z$, the encoder must invert the generator at that $z$, such that $E(G(z))=z$, which is exactly $L_{g}$ item in TEM doing!

 - At that moment, generating high-resolution images remains difficult for generative models. Thus, the encoder may take higher resolution input while the generator output and discriminator input remain low resolution.

**Comment by NorbertZheng** [comment]
### PASCAL VOC
![image](https://user-images.githubusercontent.com/37235023/216503107-9d21fdf6-3686-4333-97b4-0703df8ef91c.png)
Classification and [Fast R-CNN](https://medium.com/coinmonks/review-fast-r-cnn-object-detection-a82e172e87ba?source=post_page---------------------------) detection results for the PASCAL VOC 2007 test set and [FCN](https://towardsdatascience.com/review-fcn-semantic-segmentation-eb8c9b50d2d1?source=post_page---------------------------) segmentation results on the PASCAL VOC 2012 validation set.
 - The transferability of BiGAN representations to the PASCAL VOC is evaluated.
 - Classification models are trained with various portions of the [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160?source=post_page---------------------------) model frozen.
 - In the fc8 column, only the linear classifier (a multinomial logistic regression) is learned — in the case of BiGAN, on top of randomly initialized fully connected (FC) layers fc6 and fc7.
 - In the fc6–8 column, all three FC layers are trained fully supervised with all convolution layers frozen.
 - Finally, in the all column, the entire network is “fine-tuned”.
 - BiGAN outperforms other unsupervised (unsup.) feature learning approaches, including the [GAN](https://medium.com/@sh.tsang/review-gan-generative-adversarial-nets-gan-e12793e1fb75)-based baselines, and despite its generality, is competitive with contemporary self-supervised (self-sup.) feature learning approaches specific to the visual domain.
 - (If interested, please read the paper for more details.)

**Comment by NorbertZheng** [comment]
## Reference
[2017 ICLR] [BiGAN] [Adversarial Feature Learning](https://arxiv.org/abs/1605.09782).

**Comment by NorbertZheng** [comment]
## Generative Adversarial Network (GAN)
Image Synthesis [[GAN](https://medium.com/@sh.tsang/review-gan-generative-adversarial-nets-gan-e12793e1fb75)] [[CGAN](https://medium.com/@sh.tsang/review-cgan-conditional-gan-gan-78dd42eee41)] [[LAPGAN](https://medium.com/@sh.tsang/review-lapgan-laplacian-generative-adversarial-network-gan-e87200bbd827)] [[AAE](https://sh-tsang.medium.com/review-aae-adversarial-autoencoders-gan-e8fda9160542)] [[DCGAN](https://medium.com/@sh.tsang/review-dcgan-deep-convolutional-generative-adversarial-network-gan-ec390cded63c)] [[CoGAN](https://sh-tsang.medium.com/review-cogan-coupled-generative-adversarial-networks-gan-273f70b340af)] [[SimGAN](https://sh-tsang.medium.com/review-simgan-learning-from-simulated-and-unsupervised-images-through-adversarial-training-gan-86a7003add50)] [[BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca)]
Image-to-image Translation [[Pix2Pix](https://sh-tsang.medium.com/review-pix2pix-image-to-image-translation-with-conditional-adversarial-networks-gan-ac85d8ecead2)] [[UNIT](https://sh-tsang.medium.com/review-unit-unsupervised-image-to-image-translation-networks-gan-4a25ced6d078)]
Super Resolution [[SRGAN & SRResNet](https://medium.com/@sh.tsang/review-srgan-srresnet-photo-realistic-super-resolution-gan-super-resolution-96a6fa19490)] [[EnhanceNet](https://medium.com/@sh.tsang/reading-enhancenet-automated-texture-synthesis-super-resolution-8429635aa75e)] [[ESRGAN](https://medium.com/towards-artificial-intelligence/reading-esrgan-enhanced-super-resolution-generative-adversarial-networks-super-resolution-e8533ad006b5)]
Blur Detection [[DMENet](https://sh-tsang.medium.com/review-dmenet-deep-defocus-map-estimation-using-domain-adaptation-blur-detection-20fdcaf5e384)]
Camera Tampering Detection [[Mantini’s VISAPP’19](https://sh-tsang.medium.com/review-mantinis-visapp-19-generative-reference-model-and-deep-learned-features-camera-f608371c9854)]
Video Coding [[VC-LAPGAN](https://medium.com/@sh.tsang/reading-vc-lapgan-video-coding-oriented-laplacian-pyramid-of-generative-adversarial-networks-74daa2d23d3c)] [[Zhu TMM’20](https://sh-tsang.medium.com/review-zhu-tmm20-generative-adversarial-network-based-intra-prediction-for-video-coding-c8a217c564ea)] [[Zhong ELECGJ’21](https://sh-tsang.medium.com/review-zhong-elecgj21-a-gan-based-video-intra-coding-hevc-intra-9e3486dbca78)]
